File: LongChallenge_2019_01_FANCY.py
# code_report Solution
# Video link: https://youtu.be/xZKd3lkYLdE
# Problem Link: https://www.codechef.com/JAN19A/problems/FANCY

# BETTER SOLUTION:

t = int(input())
for _ in range(t):
    s = set(input().strip().split())
    if 'not' in s: print('Real Fancy') 
    else:          print('regularly fancy')

# BAD SOLUTION

def isFancy(s):
    n = len(s)
    return s == 'not' or \
           (n >= 4 and (s[:4]  == 'not ' or \
                        s[-4:] == ' not' or \
                        ' not ' in s))
                    
t = int(input())
for _ in range(t):
    s = input().strip()
    print('Real Fancy') if isFancy(s) else print('regularly fancy')

File: LongChallenge_2019_01_FANCY.cpp
// code_report Solution
// Video link: https://youtu.be/xZKd3lkYLdE
// Problem Link: https://www.codechef.com/JAN19A/problems/FANCY

#include <iostream>
#include <string>

using namespace std;

auto is_fancy(const string& s) {
   auto n = s.size();
   return s == "not" ||
            (n >= 4 && (s.substr(0, 4)     == "not " ||
                        s.substr(n - 4, 4) == " not" ||
                        s.find(" not ") != string::npos));
}

int main() {
   int t; cin >> t;
   string s; getline(cin, s);
   while (t--) {
      string s; getline(cin, s);
      cout << (is_fancy(s) ? "Real Fancy\n" : "regularly fancy\n");
   }
   return 0;
}
